### ISSUE TITLE
Slowly deleting many snapshots should display a progress indicator

### ISSUE THREAD

**Issue by jondo** [bit-team/backintime#1264]

**Comment by emtiu** [comment]
I'm merging Issues #237, #707 and #802 into this one. Here's a summary of the information contained there (as well as in #152, which is related but not a duplicate):

* In past versions (before 1.1.10), removing snapshots froze the GUI until the deletion was complete. This was reported in #152 and #487, and fixed with 0629361c16a053ab1baa26d607edca2943072fe8 and bc6d6bdf0309c94f7e19783a6d3f2d877a224074 by introducing: https://github.com/bit-team/backintime/blob/88d19d42e9128f60b8952a64ed1fff23bd28108f/qt/snapshotsdialog.py#L401
* Since then, snapshot deletion happens detached in the background, by spawning `find […] -exec chmod […]` and `rm […]` commands

This new behavior led to other complaints in #707, #802 and this Issue:

* Manual snapshot deletion is not represented in the status bar or the system tray. It just "happens", and users have no way to be sure if and when the process is running, stuck or finished.
  * The situation is a little different for smart remove, which at least gives a "Smart remove x/y" indicator in the status bar, spawned by (I think): https://github.com/bit-team/backintime/blob/88d19d42e9128f60b8952a64ed1fff23bd28108f/common/snapshots.py#L1378
* In https://github.com/bit-team/backintime/issues/802#issuecomment-332983410, @buhtzz described a workaround to observe snapshot deletion: _"currently I only remove snapshots when I started BIT with --debug in the bash. Only on the bash I can see for sure if the remove-task is finished or not."_

On the prospect of improving the situation and adding better status/progress indicators for snapshot deletion, longtime maintainer @Germar said:

* in https://github.com/bit-team/backintime/issues/707#issuecomment-281331073: _"The status bar currently only shows status from running snapshots. It is a relict from old times which also doesn't work very well. I want to replace it, but first I need to implement a proper interprocess communication between snapshots, GUI, systray, ... related to #84"_
* in https://github.com/bit-team/backintime/issues/802#issuecomment-333239634: _"The current way of handling the background backup jobs from detached GUI is pain in the a**. I won't do the same for background delete job. The whole IPC need to be redesigned. Because this need to work without dbus it willl be quite a big project […]"_

**In summary, the current state of affairs is:** Snapshot deletion happens detached in the background, and (in the case of manual snapshot deletion) isn't represented in the GUI in any way. The way in which the process of _taking_ a snapshot is displayed in the status bar and system tray is (according to @Germar) _not_ easily extended to _deleting_ a snapshot. In his view, the inter-process communication (IPC) between the background processes, GUI and system tray needs a complete reworking.

**Comment by Douglas1949** [comment]
Process gets stuck at "Smart remove 6/7".  Sorry, I'm 74-years-old and not tech savy. I am running Ubuntu 22.04 LTS. My Back-In-Time gets stuck at "Smart remove 6/7" - this happens even after rebooting. Can anyone suggest what I should do to rectify the situation. I should be most grateful if you could give very specific instructions as "beyond following instructions, am not tech savvy.
Many thanks, Duncan

**Comment by aryoda** [comment]
@Douglas1949
- Which BiT version are you using (`backintime --version` in the console)?
- Are you using BiT as user or root?
- Which settings do you use in your backup profile (configuration) in BiT that gets stuck? Mainly in the profiles configuration dialog I need to know
  - General > Mode (may be `local`, `ssh` ...)
  - General > Schedule (`disabled` or something else)
  - Where are the original files that are backed up stored (local disk/ssd or external disk, NAS...)
  - Attaching a screen shot of "General > Auto-remove" (showing the "smart remove" settings) would also be helpful

**Comment by Douglas1949** [comment]
Many thanks for your reply - greatly appreciated.
Here are the answers to your questions:
1) BiT version 1.2.1
2) I am root - and using as such - BUT, still get funny message every now
and then - see one of the two screenshots attached
3) Settings: Mode: local   /    Schedule disabled   /   Target files are on
pc HDD and backed up to ExtHDD   /   See screenshots
Many thanks for your time and help - I do appreciate it.
Douglas/Duncan

**Comment by Douglas1949** [comment]
I have uploaded the two screenshots to my evernote account and given the
following email address access to the file - you should have received the
invitation by now:
***@***.***

Once again - many thanks
Douglas/Duncan

**Comment by aryoda** [comment]
@Douglas1949 I have created a new issue since this issue is reserved for a special topic. Could we please continue our conversation in the new issue #1617?

I have migrated our conversation already...

> I have uploaded the two screenshots to my evernote account and given the following email address access to the file - you should have received the invitation by now: ***@***.***

I did not receive an email and I guess you cannot even see my email. Could you please upload for screenshots in the new issue #1617 in a new comment? Or in case of privacy concerns it'd suffice to write the exact wording of the "funny message" into a new comment.
